### ISSUE TITLE
Issue upgrading from 2.0.0 to 2.1.0

### ISSUE THREAD

**Issue by noftaly** [terraform-aws-modules/terraform-aws-route53#46]
## Description

Hello, I just started using this module in its latest version (2.1.0); but encountered a bug that's fixed when I use 2.0.0.
I have this setup for route53, which is not too far from the one in the examples.
```tf
module "zones" {
  source  = "terraform-aws-modules/route53/aws//modules/zones"
  version = "2.1.0"

  zones = {
    (var.account_domain_name) = {
      comment = var.account_domain_name
      tags = {
        Name = var.account_domain_name
      }
    }
  }

  tags = {
    ManagedBy = "Terraform"
  }
}

module "records" {
  source  = "terraform-aws-modules/route53/aws//modules/records"
  version = "2.1.0"

  zone_name = keys(module.zones.route53_zone_zone_id)[0]

  records = [
    {
      name    = module.instance_a.tags[0].Name # module.instance_a is an ec2 instance created with the module terraform-aws-modules/ec2-instance/aws
      type    = "A"
      ttl     = 86400  # 1 day
      records = module.instance_a.private_ip
      zone_id = module.zones.route53_zone_zone_id
    }
  ]

  depends_on = [module.zones]
}
```

The thing is, when I apply this config, it tells me:
```
╷
│ Error: Invalid for_each argument
│ 
│   on .terraform/modules/records/modules/records/main.tf line 19, in resource "aws_route53_record" "this":
│   19:   for_each = var.create && (var.zone_id != null || var.zone_name != null) ? local.recordsets : tomap({})
│     ├────────────────
│     │ local.recordsets will be known only after apply
│     │ var.create is true
│     │ var.zone_id is null
│     │ var.zone_name is "test-project.com"
│ 
│ The "for_each" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this,
│ use the -target argument to first apply only the resources that the for_each depends on.
╵
```

Again, when I downgrade to 2.0.0 I don't have this issue!

## ⚠️ Note

Before you submit an issue, please perform the following first:

1. Remove the local `.terraform` directory (! ONLY if state is stored remotely, which hopefully you are following that best practice!): `rm -rf .terraform/`
2. Re-initialize the project root to pull down modules: `terraform init`
3. Re-attempt your terraform plan or apply and check if the issue still persists

## Versions

 - Terraform: `v1.0.0`

 - Provider(s): `provider registry.terraform.io/hashicorp/aws v3.45.0`

 - Module: `2.1.0`

### Code Snippet to Reproduce
*Provided above*

## Expected behavior
I expected it to work as it used to, by seeing that my `records` object is not jsonencoded, and thus iterating over it properly.

## Actual behavior
I guess `try()` or `jsondecode()` does not behave correctly, so `locals.records` does not contain the right value.

### Terminal Output Screenshot(s)
*stdout provided above*

## Additional context
N/A

**Comment by antonbabenko** [comment]
I think this has been fixed in #54.

[v2.3.0](https://github.com/terraform-aws-modules/terraform-aws-route53/releases/v2.3.0) has been released yesterday.

Please try it and let us know if the issue is still there.

**Comment by rmasclef** [comment]
Hi,

Sorry to re-open the discussion, but it seems that the issue is still present 😢 

Let me know if you want some example to reproduce it 👍

**Comment by antonbabenko** [comment]
Yes, please open a new issue with the complete code to reproduce the issue.

**Comment by rmasclef** [comment]
I'm afraid I can't reproduce the issue with simple examples yet, though it is actually present on two projects of mine 🤯
